crates/iceberg/src/scan.rs

```diff
@@ -186,6 +240,25 @@ impl TableScan {
         .boxed())
     }
 
+    fn create_manifest_evaluator(
+        id: i32,
+        schema: SchemaRef,
+        table_metadata: Arc<TableMetadata>,
+        case_sensitive: bool,
+        filter: &Predicate,
+    ) -> crate::Result<ManifestEvaluator> {
+        let bound_predicate = filter.bind(schema.clone(), case_sensitive).unwrap();
```

Comment:
```suggestion
        let bound_predicate = filter.bind(schema.clone(), case_sensitive)?;
```

We should not panic here.